web/src/main/java/org/cbioportal/web/ClinicalAttributeController.java

```diff
@@ -114,4 +121,39 @@
         return new ResponseEntity<>(clinicalAttributeService.getClinicalAttribute(studyId, clinicalAttributeId),
                 HttpStatus.OK);
     }
+    
+    @RequestMapping(value = "/clinical-attributes/fetch", method = RequestMethod.POST,
+                    produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all clinical attributes in specified sampleIdentifiers or sampleListID with clinical attribute count")
+    public ResponseEntity<List<ClinicalAttribute>> getAllClinicalAttributesInStudies(
+    			 @ApiParam(required = false, value = "List of sample identifiers")
+             @RequestBody List<SampleIdentifier> sampleIdentifiers,
+             @ApiParam("Sample list ID e.g. acc_tcga_all")
+             @RequestParam(required = false) String sampleListId,
+             @ApiParam("Level of detail of the response")
+             @RequestParam(defaultValue = "SUMMARY") Projection projection,
+             @ApiParam("Name of the property that the result list is sorted by")
+    	         @RequestParam(required = false) ClinicalAttributeSortBy sortBy,
+    	         @ApiParam("Direction of the sort")
+    	         @RequestParam(defaultValue = "ASC") Direction direction) throws StudyNotFoundException, SampleNotFoundException, SampleListNotFoundException {
+    			
+		    	if (sampleListId != null && sampleListId.length() > 0) {
+					return new ResponseEntity<>(clinicalAttributeService.getAllClinicalAttributesInStudies(sampleListId, projection.name(), sortBy == null ? null : sortBy.getOriginalValue(), direction.name()), HttpStatus.OK);
+			}
+		    	else if (sampleIdentifiers != null && sampleIdentifiers.size() > 0) {
+    				List<String> studyIds = new ArrayList<>();
+	    			List<String> sampleIds = new ArrayList<>();
+    				for (SampleIdentifier sampleIdentifier : sampleIdentifiers) {
+    		            studyIds.add(sampleIdentifier.getStudyId());
+    		            sampleIds.add(sampleIdentifier.getSampleId());
+    		        }
+    				return new ResponseEntity<>(clinicalAttributeService.getAllClinicalAttributesInStudies(studyIds, sampleIds, projection.name(), sortBy == null ? null : sortBy.getOriginalValue(), direction.name()), HttpStatus.OK);
+
+    			} else {
```

Comment:
You won't need this when you create the request body filter object I mentioned above.